public/js/Histogram.js

```diff
@@ -0,0 +1,100 @@
+var d3 = require('d3');
+var _ = require('lodash');
+_.mixin(require('lodash-deep'));
+
+function Histogram(sampleData, element, featureNames) {
```

Comment:
btw, do you have a blog? You should: it's really great for exposure — I'm pretty sure that's how I got my book deal! Regardless, you should write a blog post that copies what you just wrote almost verbatim. =)

It is also really good practice to write often, and then you can pilfer it when you are writing your thesis!